relay-server/src/metrics_extraction/spans/mod.rs

```diff
@@ -303,6 +312,28 @@ fn sanitized_span_description(
     Some(sanitized)
 }
 
+/// Trims the given span description to ensure
+/// [`relay-metrics::AggregatorService`] doesn't drop it for being too long.
+///
+/// If the description is short, it remains unchanged. If it's long, this method
+/// truncates it to the maximum allowed size and sets the last character to
+/// `*`.
+fn truncate_span_description(mut description: String, max_bytes: usize) -> String {
+    if bytecount::num_chars(description.as_bytes()) < max_bytes {
+        return description;
+    }
+
+    let mut cutoff = max_bytes - 1; // Leave space for `*`
+
+    while cutoff > 0 && description.is_char_boundary(cutoff) {
```

Comment:
We want the boundary check negated, right? For ASCII-strings, `is_char_boundary` is always true, so this would erase the string.

```suggestion
    while cutoff > 0 && !description.is_char_boundary(cutoff) {
```